src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/arrays/ArrayFunctionIterator.java

```diff
@@ -29,7 +31,7 @@ public Item getResult() {
         // if no results return empty sequence
```

Comment:
Part of this code should go into an override of open(), in the same class (and in every iterator that has the issue). That way, _hasNext will be set to false before hasNext() is called. That should do the trick and EmptySequenceItem won't actually be returned.